src/puppetlabs/trapperkeeper/services/webserver/jetty10_core.clj

```diff
@@ -678,7 +678,14 @@
       (.stop server)
       (catch TimeoutException e
         (log/error e (i18n/trs "Web server failed to shut down gracefully in configured timeout period ({0}); cancelling remaining requests."
-                               (.getStopTimeout server)))))
+                               (.getStopTimeout server))))
+      ;; This exception handling was added since we currently manually stop handlers within pcp-broker
+      ;; for debugging purposes there - on shutdown if Jetty 10 sees a STOPPED handler it throws an
+      ;; ExecutionException with a IllegalStateExeception as it's cause and if unhandled shutdown
+      ;; stops and the server goes into a FAILED state
+      (catch ExecutionException e
+        (log/error (.getCause e) (i18n/trs "Web server failed to shut down gracefully due to ExecutionException with inner exception of type {0}; cancelling remaining requests."
```

Comment:
You can just use `e` here instead of `(.getCause e)` which will include the stack crawl as well as cascaded exceptions which can be helpful information.  Sometimes `getCause` can be null.

Reply:
That also removes the need to add the type check in the string and just make it an error message. ""Web server failed to shut down gracefully"